standalone-tools/wss-scan.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+###### Information ############################################################################
+# Name:          wss-scan.sh
+# Maintainer:    ODFE Infra Team
+# Language:      Shell
+#
+# About:         This script is to scan the odfe distros for vulnerabilities and licenses
+#                It will scan the repositories and send the WhiteSource link to the mail 
+#                of the user. 
+#
+# Prerequisites: Need to install Java 11
+#                Export JAVA_HOME env variable to the JDK path
+#		 Add JAVA_HOME to PATH variable
+#                Need to set the recepient mail in wss-scan.config for local run
+#                WhiteSource API key is needed for local run, The API Key can be retrieved from the
+#                WhiteSource Admin Console of your account. Use the below command to export the API key
+#                export wss_apikey=$(APIKEY)
+#
+# Usage:         ./wss-scan.sh
+#
+# Starting Date: 10-13-2020
+# Modified Date: 10-21-2020
+###############################################################################################
+
+set -e
+
+java -version
+if [ "$?" != 0 ]
+then 
+  echo "Java has not been setup" 
+  exit 1
+fi
+
+if [ ! -f "wss-unified-agent.jar" ]
+then
+  # Download the WhiteSource Agent 
+  wget -q https://github.com/whitesource/unified-agent-distribution/releases/latest/download/wss-unified-agent.jar
+  # The version 20.9.2.1 has been tested and can be used if a specific version is required
+  #curl -LJO -sS https://github.com/whitesource/unified-agent-distribution/releases/download/v20.9.2.1/wss-unified-agent.jar
+fi
+
+# scan the config file for the user configurations
+# wss-scan.config has to be present in the same working directory as the script
+source wss-scan.config
+
+# change comma to whitespace
+gitRepos=${gitRepos//,/$'\n'} 
+
+basepath=$baseDirPath"/repos"
+
+echo "Cleaning up scan directories if already present"
+rm -rf $basepath
+
+mkdir -p $basepath
+ 
+
+# clone the desired Repos for scanning 
+for repo in $gitRepos
+do
+  echo "Cloning repo "$gitBasePath$repo
+  git clone "$gitBasePath$repo".git $basepath"/"$repo
+done
+
+echo -n > info.txt
+
+# scan the Repos using the WhiteSource Unified Agent
+for repo in $gitRepos
+do
+  repo_path=$basepath"/"$repo
+  if [ -d "$repo_path" ]
+  then
+    echo "Scanning repo: "$gitBasePath$repo " Project: " $repo 
+    java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product ODFE -project $repo | grep "Project name" | sed 's/^.\{,41\}//' >> info.txt 2>&1 &
+  else
+    echo "Scanning failed for repo: "$gitBasePath$repo " Project: " $repo
+  fi
+done
+
+
+i=0
+# wait for the scannings to complete
+while ps ax | grep -wq [w]ss-unified-agent.jar
+do
+  echo "Scanning is still in progress"
+  sleep 60
+  ((i=i+1))
+  # break the loop after 70 mins
+  if [ "$i" -gt 70 ] 
+  then
+    break 
```

Comment:
Can you fail the script here with an error message rather than continuing?

Reply:
If I fail the script, then all the other scan information will also be lost. If there is process which has not completed, I would just skip it and send the mail with the remaining scan results.